packages/ui/__tests__/Tooltip.test.tsx

```diff
@@ -1,64 +1,107 @@
 
+/**
+ * There's a weird issue with react-popper that results in missing act() wrapper error.
+ * This is why all mounts are wrapped in act().
+ *
+ * https://github.com/popperjs/react-popper/issues/368
+ */
 describe('Tooltip', () => {
-  test('renders children without Tooltip if "overlay" is undefined', () => {
-    const Child = () => <div />
+  test('Renders correctly if "content" property is defined.', async () => {
+    await act(async () => {
+      const wrapper = mount(
+        <Tooltip id="tooltip-test" content="Tooltip content">
+          {(ref) => (
+            <button ref={ref} data-test="tooltip-reference">
+              Hover me
+            </button>
+          )}
+        </Tooltip>,
+      )
+
+      const tooltipOverlay = wrapper.find('[data-test="tooltip-overlay"]')
+
+      expect(wrapper.find('[data-test="tooltip-reference"]').exists()).toBeTruthy()
+      expect(tooltipOverlay.exists()).toBeTruthy()
+      expect(tooltipOverlay.text()).toEqual('Tooltip content')
+    })
+  })
+
+  test('Does not render tooltip if "content" property is not defined.', async () => {
+    await act(async () => {
+      const wrapper = mount(
+        <Tooltip id="tooltip-test" content={undefined}>
+          {(ref) => (
+            <button ref={ref} data-test="tooltip-reference">
+              Hover me
+            </button>
+          )}
+        </Tooltip>,
+      )
+
+      expect(wrapper.find('[data-test="tooltip-overlay"]').exists()).toBeFalsy()
+
+      wrapper.find('[data-test="tooltip-reference"]').simulate('mouseenter')
+
+      wrapper.update()
+
+      expect(wrapper.find('[data-test="tooltip-overlay"]').exists()).toBeFalsy()
+    })
+  })
+
+  test('Shows tooltip overlay on hover of the target element.', async () => {
+    await act(async () => {
+      const wrapper = mount(
+        <Tooltip id="tooltip-test" content="Tooltip content">
+          {(ref) => (
+            <button ref={ref} data-test="tooltip-reference">
+              Hover me
+            </button>
+          )}
+        </Tooltip>,
+      )
+
+      let tooltipOverlay = wrapper.find('[data-test="tooltip-overlay"]')
+
+      expect(tooltipOverlay.hasClass('hidden')).toBeTruthy()
+
+      wrapper.find('[data-test="tooltip-reference"]').simulate('mouseenter')
 
-    const wrapper = mount(
-      <Tooltip overlay={undefined}>
-        <Child />
-      </Tooltip>,
-    )
+      wrapper.update()
 
-    expect(wrapper.exists(RCTooltip)).toBeFalsy()
-    expect(wrapper.exists(Child)).toBeTruthy()
+      tooltipOverlay = wrapper.find('[data-test="tooltip-overlay"]')
+
+      expect(tooltipOverlay.hasClass('hidden')).toBeTruthy()
+      expect(tooltipOverlay.text()).toEqual('Tooltip content')
+      expect(tooltipOverlay.prop('id')).toEqual('tooltip-test')
+
+      wrapper.find('[data-test="tooltip-reference"]').simulate('mouseleave')
+
+      setTimeout(() => {
+        wrapper.update()
+        expect(wrapper.find('[data-test="tooltip-overlay"]')).toBeFalsy()
```

Comment:
Shouldn't we expect ti to be `true` here?

Reply:
No longer relevant:
https://github.com/hazelcast/frontend-shared/pull/27#discussion_r514193417